prow/plugins/warn-owners-aliases/warn-owners-aliases.go

```diff
@@ -0,0 +1,177 @@
+	base         github.PullRequestBranch
+	head         github.PullRequestBranch
+	number       int
+	org          string
+	repo         string
+	repoFullName string
+	user         string
+}
+
+func handlePullRequest(pc plugins.Agent, pre github.PullRequestEvent) error {
+	// Only consider newly opened PRs
+	if pre.Action != github.PullRequestActionOpened {
+		return nil
+	}
+
+	prInfo := info{
+		base:         pre.PullRequest.Base,
+		head:         pre.PullRequest.Head,
+		number:       pre.Number,
+		org:          pre.Repo.Owner.Login,
+		repo:         pre.Repo.Name,
+		repoFullName: pre.Repo.FullName,
+		user:         pre.PullRequest.User.Login,
+	}
+
+	return handle(pc.GitHubClient, pc.OwnersClient, pc.Logger, prInfo, pc.PluginConfig.OwnersFilenames)
+}
+
+func handle(ghc githubClient, oc ownersClient, log *logrus.Entry, info info, resolver ownersconfig.Resolver) error {
+	// Get changes.
+	changes, err := ghc.GetPullRequestChanges(info.org, info.repo, info.number)
+	if err != nil {
+		return fmt.Errorf("error getting PR changes: %w", err)
+	}
+
+	// Check if OWNERS_ALIASES has been modified.
+	var ownersAliasesModified bool
+	filenames := resolver(info.org, info.repo)
+	for _, change := range changes {
+		if change.Filename == filenames.OwnersAliases &&
+			change.Status != github.PullRequestFileRemoved {
+			ownersAliasesModified = true
+			break
+		}
+	}
+	if !ownersAliasesModified {
+		return nil
+	}
+
+	log.Debug("Resolving repository owners for base branch...")
+	baseRepo, err := oc.LoadRepoOwnersSha(info.org, info.repo, info.base.Ref, info.base.SHA, false)
+	if err != nil {
+		return err
+	}
+	// This shouldn't happen, but if the repo had no owners
+	// aliases defined, don't do anything and return.
+	if len(baseRepo.OwnersAliases()) == 0 {
+		return nil
+	}
+
+	log.Debug("Resolving repository owners for head branch...")
+	headRepo, err := oc.LoadRepoOwnersSha(info.org, info.repo, info.head.Ref, info.head.SHA, false)
+	if err != nil {
+		return err
+	}
+	// If for some reason all owners aliases were deleted as part of this PR,
+	// don't do anything and return.
+	if len(headRepo.OwnersAliases()) == 0 {
+		return nil
+	}
+
+	aliasesAddedTo := findOwnersAliasesAddedTo(baseRepo.OwnersAliases(), headRepo.OwnersAliases())
+	// if no relevant changes, don't do anything.
+	if len(aliasesAddedTo) == 0 {
+		return nil
+	}
+
+	return ghc.CreateComment(info.org, info.repo, info.number, constructComment(info.org, info.repo, info.user, aliasesAddedTo))
+}
+
+// findOwnersAliasesAddedTo finds the existing aliases that were added to
+// by comparing the lengths of the owners under each alias.
+func findOwnersAliasesAddedTo(base, head repoowners.RepoAliases) []string {
+	result := []string{}
+	for alias := range head {
+		owners, ok := base[alias]
+		if !ok { // newly added alias.
+			continue
+		}
+		// If someone was added to this already existing alias
+		// as part of this PR, return this alias.
+		if head[alias].Difference(owners).Len() > 0 {
```

Comment:
@MadhavJivrajani, `if head[alias].Difference(owners)` is handling which of the following?

(i) new owners that are added as part of the PR? OR
(ii) all from (i) +  also return the owners that a PR might be removing?

Reply:
So this handles only additions. If there are removals, the result of `.Len()` will be 0.

Also - we don't really need to worry about removals here since the case of elevated access don't really come into play in that case.